app/mailers/host_mailer.rb

```diff
@@ -6,6 +6,7 @@ class HostMailer < ApplicationMailer
   # sends out a summary email of hosts and their metrics (e.g. how many changes failures etc).
   def summary(options = {})
     raise ::Foreman::Exception.new(N_("Must specify a valid user with email enabled")) unless (user = User.find(options[:user]))
+    User.current = user
```

Comment:
I think this should be done in a block, possibly via `User.as user do ... end` to make sure the user context is switched back.